src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1287 @@
+        UfbxMaterialLayerAttributes& attributesForLayer = layerAttributes[UnsignedInt(layer)];
+
+        /* Skip empty layers after the first one */
+        if(layer != 0 && attributesForLayer.defaultLayer.isEmpty()) continue;
+
+        /* Default layer within the named layer */
+        {
+            UnsignedInt layerAttributeCount = 0;
+            Containers::ArrayView<MaterialAttributeData> attributes = attributesForLayer.defaultLayer;
+            if(layer != 0) {
+                arrayAppend(flatAttributes, {MaterialAttribute::LayerName, ufbxMaterialLayerNames[layer]});
+                ++layerAttributeCount;
+            }
+            arrayAppend(flatAttributes, attributes);
+            layerAttributeCount += attributes.size();
+
+            layerOffset += layerAttributeCount;
+            arrayAppend(layerSizes, layerOffset);
+        }
+
+        /* Extra layers (ie. extra texture layers in FBX) */
+        for(UnsignedInt i = 0; i < attributesForLayer.extraLayers.size(); ++i) {
+            UnsignedInt layerAttributeCount = 0;
+            Containers::ArrayView<MaterialAttributeData> attributes = attributesForLayer.extraLayers[i];
+            if(layer != 0) {
+                arrayAppend(flatAttributes, {MaterialAttribute::LayerName, ufbxMaterialLayerNames[layer]});
+                ++layerAttributeCount;
+            }
+            arrayAppend(flatAttributes, attributes);
+            layerAttributeCount += attributes.size();
+
+            layerOffset += layerAttributeCount;
+            arrayAppend(layerSizes, layerOffset);
+        }
+    }
+
+    /* Convert back to the default deleter to avoid dangling deleter function
+       pointer issues when unloading the plugin */
+    arrayShrink(flatAttributes, DefaultInit);
+    arrayShrink(layerSizes, DefaultInit);
+
+    return MaterialData{types, std::move(flatAttributes), std::move(layerSizes)};
+}
+
+UnsignedInt UfbxImporter::doTextureCount() const {
+    return UnsignedInt(_state->textures.size());
+}
+
+Int UfbxImporter::doTextureForName(Containers::StringView name) {
+    const ufbx_element* element = ufbx_find_element_len(_state->scene.get(), UFBX_ELEMENT_TEXTURE, name.data(), name.size());
+    return element ? _state->textureRemap[element->typed_id] : -1;
+}
+
+Containers::String UfbxImporter::doTextureName(UnsignedInt id) {
+    const FileTexture& fileTexture = _state->textures[id];
+    return _state->scene->textures[fileTexture.textureIndex]->name;
+}
+
+Containers::Optional<TextureData> UfbxImporter::doTexture(UnsignedInt id) {
+    const FileTexture& fileTexture = _state->textures[id];
+    const ufbx_texture* texture = _state->scene->textures[fileTexture.textureIndex];
+
+    const SamplerWrapping wrappingU = toSamplerWrapping(texture->wrap_u);
+    const SamplerWrapping wrappingV = toSamplerWrapping(texture->wrap_v);
+    return TextureData{TextureType::Texture2D,
+        SamplerFilter::Linear, SamplerFilter::Linear, SamplerMipmap::Linear,
```

Comment:
Just to be sure we're not missing anything here -- Assimp doesn't really have texture filtering properties (apart from a glTF-specific `AI_MATKEY_GLTF_MAPPINGFILTER_*`), so this was hardcoded to the "best available" filter mode. 

How is it with FBX, is it also assuming linear filtering for everything? A comment would make sense I think, possibly also a section about texture properties in the public docs in the header.

Reply:
Good point with documenting it, afaik FBX doesn't specify anything about texture filtering (except 3ds Max has probably found a proprietary-within-proprietary-format way of doing it again..)